core/domain/suggestion_services.py

```diff
@@ -542,3 +555,43 @@ def check_can_resubmit_suggestion(suggestion_id, user_id):
     suggestion = get_suggestion_by_id(suggestion_id)
 
     return suggestion.author_id == user_id
+
+
+def check_can_edit_suggestion(user, suggestion_id):
+    """Checks whether the user have permission to edit suggestion or not.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        suggestion_id: str. Id of a suggestion.
+
+    Returns:
+        users_which_have_rights: dict(str, str). Dict whose key value pairs
+            describes whether a user have permisson or not.
+    """
+
+    suggestion = get_suggestion_by_id(suggestion_id)
+    user_id = user.user_id
+
+    if suggestion.author_id == user_id:
+        return True
+
+    if suggestion.target_type == 'exploration':
+        exploration_rights = rights_manager.get_exploration_rights(
+            suggestion.target_id)
+        if exploration_rights is None:
+            return False
+
+        if rights_manager.check_can_edit_activity(
+                user, exploration_rights):
+            return True
+
+    elif suggestion.target_type == 'collection':
```

Comment:
Currently target_type has only been implemented for exploration and topic. It still hasn't been implemented for collections. Keeping this is fine, but make sure to add a case for topics.

Reply:
You missed this comment, I think? Let me know if you want to discuss this.

Reply:
Sorry for this, i forget about this.
Done PTAL.